DOC: More details about creating Azure resources (#498)
microsoft/hi-ml: docs/source/azure_setup.md

# Setting up Azure

If you already have an AzureML workspace available, you can go straight to [the last step](#accessing-the-workspace).

To set up Azure, you need to have:

- A subscription to Azure.
- An account that has "Owner" permissions to the Azure subscription.

There are two ways to set up all necessary resources, either via the Azure portal or via the Azure Commandline Interface (CLI).
We recommend the CLI because all necessary resources can be easily created via a single script.

## Creating an AzureML workspace via the Azure Portal

If you prefer to create your workspace via the web UI on the Azure Portal, please follow the steps below.

- [Create a resource

>   group](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)

Reviewer: ```suggestion
  group.](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/manage-resource-groups-portal)
```

- [Create a storage account for

>   datasets](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal)

Reviewer: ```suggestion
  datasets.](https://docs.microsoft.com/en-us/azure/storage/common/storage-account-create?tabs=azure-portal)
```

> - [Create an AzureML workspace, compute instances and compute clusters](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources)

Reviewer: ```suggestion
- [Create an AzureML workspace, compute instances and compute clusters.](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources)
```

## Creating an AzureML workspace via the Azure Commandline Tools

> A pureley commandline-drive setup is possible via the [Azure Commandline Tools](https://docs.microsoft.com/en-us/cli/azure/). These tools are available for multiple platforms, including Linux, Mac, and Windows.

Reviewer: ```suggestion
A purely command line-drive setup is possible via the [Azure Command Line Tools](https://docs.microsoft.com/en-us/cli/azure/). These tools are available for multiple platforms, including Linux, Mac, and Windows.
```

> After downloading the commandline tools, you can run the following command add the `ml` extension that is required to create an AzureML workspace:

Reviewer: ```suggestion
After downloading the commandline tools, you can run the following command to add the `ml` extension that is required to create an AzureML workspace:
```

```bash
az extension add --name ml
```

### Documentation

> - [Creating AzureML workspaces via the CLI](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-workspace-cli)

Reviewer: ```suggestion
- [Creating AzureML workspaces via the CLI.](https://docs.microsoft.com/en-us/azure/machine-learning/how-to-manage-workspace-cli)
```

> - [Creating Azure storage accounts via the CLI](https://docs.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)

Reviewer: ```suggestion
- [Creating Azure storage accounts via the CLI.](https://docs.microsoft.com/en-us/cli/azure/storage?view=azure-cli-latest)
```

> - [Schema for creating datastores](https://docs.microsoft.com/en-us/azure/machine-learning/reference-yaml-datastore-blob)

Reviewer: ```suggestion
- [Schema for creating datastores.](https://docs.microsoft.com/en-us/azure/machine-learning/reference-yaml-datastore-blob)
```

- [Creating datastores via the CLI](https://docs.microsoft.com/en-us/cli/azure/ml/datastore?view=azure-cli-latest)
- [Create a shared access signature](https://docs.microsoft.com/en-us/azure/storage/blobs/storage-blob-user-delegation-sas-create-cli)

### Collecting the necessary information

> Find out which Azure datacenter locations you can use:

Reviewer: ```suggestion
Find out which Azure data centre locations you can use:
```

    --resource-group ${prefix}rg \
    --location ${location} \
    --sku Standard_LRS
az storage container create \
    --account-name ${prefix}data \
    --name ${container} \
    --auth-mode
az ml workspace create \
    --resource-group ${prefix}rg \
    --name ${prefix} \
    --location ${location}
key=$(az storage account keys list --resource-group ${prefix}rg --account-name ${prefix}data --query [0].value -o tsv)cat >${datastorefile} <<EOL
\$schema: https://azuremlschemas.azureedge.net/latest/azureBlob.schema.json
name: datasets
type: azure_blob
description: Pointing to the datasets container in the ${prefix}data storage account.
account_name: ${prefix}data
container_name: ${container}
credentials:
  account_key: ${key}
EOL
az ml datastore create --file ${datastorefile} --resource-group ${prefix}rg --workspace-name ${prefix}
rm ${datastorefile}
```

Note that the datastore will use the storage account key to authenticate. You may want to switch that to Shared Access Signature (SAS).

Now that you have created the core AzureML workspace, you need to
[create a compute cluster](https://docs.microsoft.com/en-us/azure/machine-learning/quickstart-create-resources#cluster).

> The final step is to download the workspace configuration file, as descreibed [here](#accessing-the-workspace).

Reviewer: ```suggestion
The final step is to download the workspace configuration file, as described [here](#accessing-the-workspace).
```
